cucumber-picocontainer/src/main/java/io/cucumber/picocontainer/PicoFactory.java

```diff
@@ -14,7 +15,14 @@
 public final class PicoFactory implements ObjectFactory {
 
     private final Set<Class<?>> classes = new HashSet<>();
-    private MutablePicoContainer pico;
+    private final MutablePicoContainer pico = new PicoBuilder()
+            .withCaching()
+            .withLifecycle()
+            .build();
+
+    public PicoFactory() {
+        this.pico.start();
```

Comment:
This will break step definitions that implement the `Startable` because the factory is no longer stopped/started at the right moment..

Reply:
What about doing the `pico.start()` only once in `PicoFactory.start()` ?

```java

private boolean firstTime = true;

public void start() {
    if (firstTime) {
      this.pico.start();
      firstTime = false;
    }
}
```

Reply:
And it doesn't because flushing from cache also invokes the life cycle methods.